docs/source/spark_rapids_ml.rst

```diff
@@ -96,4 +98,17 @@ Tuning
     :template: autosummary/class_with_docs.rst
     :toctree: api
 
-    CrossValidator
\ No newline at end of file
+    CrossValidator
+    
+
+UMAP
+----
+
+.. currentmodule:: spark_rapids_ml.umap
+
+.. autosummary::
+    :template: autosummary/class_with_docs.rst
+    :toctree: api
+
+    UMAP
+    UMAPModel
```

Comment:
Might be able to add the following to `feature.py`:
```
from .umap import UMAP, UMAPModel
```

But not sure how this would affect the doc generation.  Not a huge deal, since it's not officially in Spark ML.

Reply:
That actually (sort of) works.  Docs generated ok.   However the docs then show the class fqn as spark_rapids_ml.feature.UMAP while all examples import as spark_rapids_ml.umap.UMAP (which we can change likely).    
However,
`from spark_rapids_ml.feature import UMAP
type(UMAP)
`
shows <class 'spark_rapids_ml.umap.UMAP'>

Reply:
Think it's fine to defer for now.